samples/src/Universal/Aspects/BaseUniversalWrapperAspect.cs

```diff
@@ -15,7 +16,7 @@ public abstract class BaseUniversalWrapperAspect
         private delegate object Wrapper(Func<object[], object> target, object[] args);
         private delegate object Handler(Func<object[], object> next, object[] args, AspectEventArgs eventArgs);
 
-        private static readonly Dictionary<MethodBase, Handler> _delegateCache = new Dictionary<MethodBase, Handler>();
+        private static readonly ConcurrentDictionary<(MethodBase, Type), Lazy<Handler>> _delegateCache = new ConcurrentDictionary<(MethodBase, Type), Lazy<Handler>>();

```

Comment:
Are you sure typeinfo is enough?

Reply:
A method handler can be reused with different constructed method types as long as the return type doesn't change, so using target.Method would allow the cache to contain interchangeable/redundant handlers.

The reason is that CreateMethodHandler() builds handlers from WrapSync<T>() and WrapAsync<T>() from BaseUniversalWrapperAttribute, where the only generic argument is the return type. (Those methods have to use an "object[] args" param, since C# lacks variadic generics.)

Reply:
this makes sense, well done